inc/classes/class-imagify-admin-ajax-post.php

```diff
@@ -375,132 +438,80 @@ public function imagify_bulk_upload_callback() {
-		$context = imagify_sanitize_context( filter_input( INPUT_POST, 'context', FILTER_SANITIZE_STRING ) );
-		$context = ! $context || 'wp' === strtolower( $context ) ? 'File' : $context;
-
-		if ( ! $file_id ) {
-			imagify_die( __( 'Invalid request', 'imagify' ) );
-		}
-
-		$file = get_imagify_attachment( $context, $file_id, 'imagify_bulk_optimize_file' );
-
-		if ( ! $file->is_valid() ) {
-			imagify_die( __( 'Invalid file ID', 'imagify' ) );
-		}
-
-		// Restore before re-optimizing.
-		if ( false !== $file->get_optimization_level() ) {
-			$file->restore();
-		}
-
-		// Optimize it.
-		$result = $file->optimize( $this->get_optimization_level() );
-
-		// Return the optimization statistics.
-		if ( ! $file->is_optimized() ) {
-			$data = array(
-				'success'    => false,
-				'error_code' => '',
-				'error'      => (string) $file->get_optimized_error(),
-			);
-
-			if ( ! $file->has_error() ) {
-				$data['error_code'] = 'already-optimized';
-			} else {
-				$message = 'You\'ve consumed all your data. You have to upgrade your account to continue';
-
-				if ( $data['error'] === $message ) {
-					$data['error_code'] = 'over-quota';
-				}
-			}
-
-			$data['error'] = imagify_translate_api_message( $data['error'] );
-
-			imagify_die( $data );
-		}
+		$media_id = $this->get_media_id( 'GET', 'id' );
 
-		$data = $file->get_size_data();
-
-		wp_send_json_success( array(
-			'success'                     => true,
-			'original_size_human'         => imagify_size_format( $data['original_size'], 2 ),
-			'new_size_human'              => imagify_size_format( $data['optimized_size'], 2 ),
-			'overall_saving'              => $data['original_size'] - $data['optimized_size'],
-			'overall_saving_human'        => imagify_size_format( $data['original_size'] - $data['optimized_size'], 2 ),
-			'original_overall_size'       => $data['original_size'],
-			'original_overall_size_human' => imagify_size_format( $data['original_size'], 2 ),
-			'new_overall_size'            => $data['optimized_size'],
-			'percent_human'               => $data['percent'] . '%',
-			'thumbnails'                  => $file->get_optimized_sizes_count(),
-		) );
+		$this->optimize_media( $media_id, 'custom-folders' );
 	}
 
 	/**
-	 * Optimize a file.
+	 * Re-optimize a file.
 	 *
 	 * @since  1.7
 	 * @access public
 	 * @author Grégory Viguier
 	 */
-	public function imagify_optimize_file_callback() {
-		imagify_check_nonce( 'imagify_optimize_file' );
+	public function imagify_reoptimize_file_callback() {
+		imagify_check_nonce( 'imagify_reoptimize_file' );
 		imagify_check_user_capacity( 'optimize-file' );
 
-		$file = $this->get_file_to_optimize( 'imagify_optimize_file' );
-
-		// Optimize it.
-		$result = $file->optimize();
+		$media_id = $this->get_media_id( 'GET', 'id' );
+		$level    = $this->get_optimization_level( 'GET', 'level' );
+		$process  = imagify_get_optimization_process( $media_id, 'custom-folders' );
```

Comment:
You set `$process`, but don't use it?

Reply:
This will change in a future branch, `$process` won't be set.